app/Http/Controllers/RulesController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Models\City;
+use App\Models\Country;
+use App\Models\Rules;
+use App\Services\OpenAIService;
+use Exception;
+
+class RulesController
+{
+    public function getRules(string $countryName, string $cityName): array
```

Comment:
Look at `CityPageController`. There's similar logic and there's no need to use try-catch.